server/src/main/scala/org/http4s/server/ServerApp.scala

```diff
@@ -34,8 +34,8 @@ trait ServerApp extends ProcessApp {
   def shutdown(server: Server): Task[Unit] =
     server.shutdown
 
-  final def process(args: List[String]): Process[Task, Unit] =
+  final def process(args: List[String]): Process[Task, Nothing] =
     Process.bracket(server(args))(s => Process.eval_(s.shutdown)) { s =>
-      Process.eval_(Task.async[Unit](_ => ()))
+      Process.eval_(Task.async[Nothing](_ => ()))
```

Comment:
This is pretty common in our code and I'd love to give it a name.   It runs forever and emits never.

I don't want to bikeshed this one too much, because our 0.17 template is broken out of the box.  I want to get this merged to master quickly.